### PR TITLE
Oidc provideruri test

### PR DESCRIPTION
**Pull request by nessiLahav**
#### What does this PR do?
Add Provider uri Dynamic change automated test 
#### What ticket does this PR close?
Connected to #1028 
#### Where should the reviewer start?
Feature file
#### How should this be manually tested?
Run OIDC cucumber tests
#### Has the Version and Changelog been updated?

> Can we make a blog post, video, or animated GIF of this?

> Has this change been documented (Readme, docs, etc.)?

> Does the knowledge base need an update?

### CODE REVIEW

File: cucumber/authenticators/features/authn_oidc.feature [hunk 1]
```diff
@@ -52,7 +52,7 @@ Feature: Users can authneticate with OIDC authenticator
       role: !group conjur/authn-oidc/keycloak/users
       member: !user alice@conjur.net
     """
-    When I add the secret value "email" to the resource "cucumber:variable:conjur/authn-oidc/keycloak/id-token-user-property"
+    When I add the secret value "email" to the resource "cucumber:variable:conjur/authn-oidc/keycloak/provider-uri"
```
Comment (orenbm) [inline cucumber/authenticators/features/authn_oidc.feature]: what's going on here? why are you adding `email` to the `provider-uri` variable?

File: cucumber/authenticators/features/authn_oidc.feature [hunk 2]
```diff
@@ -157,6 +157,31 @@ Feature: Users can authneticate with OIDC authenticator
     Errors::Authentication::AuthnOidc::AdminAuthenticationDenied
     """
 
+  Scenario: Provider uri Dynamic change
```
Comment (orenbm) [inline cucumber/authenticators/features/authn_oidc.feature]: Please rename to "provider-uri variable dynamic change"

File: cucumber/authenticators/features/authn_oidc.feature [hunk 3]
```diff
@@ -157,6 +157,31 @@ Feature: Users can authneticate with OIDC authenticator
...
+    And The following appears in the log after my savepoint:
+    """
+    504 Gateway Timeout
+    """
+    # Update provider uri to reachable but invalid hostname
+    When I add the secret value "http://127.0.0.1.com/" to the resource "cucumber:variable:conjur/authn-oidc/keycloak/provider-uri"
+    And I authenticate via OIDC with id token
+    Then it is bad gateway
+    # Check recovery to a valid provider uri
+    Given I successfully set OIDC variables
```
Comment (orenbm) [inline cucumber/authenticators/features/authn_oidc.feature]: this is not given. it looks like you're doing something so it should be `when`

File: cucumber/authenticators/features/authn_oidc.feature [hunk 4]
```diff
@@ -157,6 +157,31 @@ Feature: Users can authneticate with OIDC authenticator
     Errors::Authentication::AuthnOidc::AdminAuthenticationDenied
     """
 
+  Scenario: Provider uri Dynamic change
+    Given I get authorization code for username "alice" and password "alice"
+    And I fetch an ID Token
+    When I authenticate via OIDC with id token
+    Then user "alice" is authorized
+    # Update provider uri to an unreachable hostname
+    When I add the secret value "http://unreachable.com/" to the resource "cucumber:variable:conjur/authn-oidc/keycloak/provider-uri"
```
Comment (orenbm) [inline cucumber/authenticators/features/authn_oidc.feature]: all the steps above this one should be given. you start the actual test here.

File: cucumber/authenticators/features/step_definitions/authn_common_steps.rb [hunk 5]
```diff
@@ -17,3 +17,11 @@
 Then(/it is forbidden/) do
   expect(forbidden?).to be true
 end
+
+Then(/it is read timeout/) do
```
Comment (orenbm) [inline cucumber/authenticators/features/step_definitions/authn_common_steps.rb]: what is "read timeout"?

File: cucumber/authenticators/features/support/authenticator_helpers.rb [hunk 6]
```diff
@@ -53,6 +53,14 @@ def forbidden?
     http_status == 403
   end
 
+  def bad_gateway?
+    http_status == 502
+  end
+
+  def read_timeout?
```
Comment (orenbm) [inline cucumber/authenticators/features/support/authenticator_helpers.rb]: i would change this method name. the behaviour of the rest of these method i to check the http status but here it is different and misleading.

File: cucumber/authenticators/features/step_definitions/authn_common_steps.rb (hunk 5 above)
Comment (nessiLahav) [reply cucumber/authenticators/features/step_definitions/authn_common_steps.rb]: The scenario here is the client (cucumber) gets the timeout before conjur (still stuck to resolve unreachable hostname), so the client gets read timeout error and null in the http returned code, so this is the way to validate the client gets timeout error

File: cucumber/authenticators/features/authn_oidc.feature (hunk 4 above)
Comment (nessiLahav) [reply cucumber/authenticators/features/authn_oidc.feature]: Done :)

**Pull request merged**